### PR TITLE
perf(sqlite): route store_prekeys_batch through the shared single-transaction write path

### PR DESCRIPTION
**Pull request by jlucaso1**
Audit finding: `put/delete_sessions_batch` and `remove_prekeys_batch` were already one transaction per batch via `with_retry`, and all live session writes already go through the batched flush — `store_prekeys_batch` was the one named batch fn still hand-rolling its own permit/`spawn_blocking`/retry loop. It now shares the exact shape (one pool checkout, one `spawn_blocking`, one transaction per attempt), with identical semantics (same upsert, empty short-circuit, same 6-attempt budget) plus the shared helper's warn logging on busy/locked retries.

Also adds the missing round-trip test for this fn (store → re-store upsert → batch remove → empty short-circuit).

Honest accounting: no material movement expected on any `signal_flush` row — the unification keeps the identical measured shape, and the fn appears in benches only in setup. This is hygiene that keeps the next flush-path change honest.

Verified: `whatsapp-rust-sqlite-storage` lib 92/92 (1 new), `fmt`/`clippy` clean. Numbers from CI.

### CODE REVIEW

File: storages/sqlite-storage/src/sqlite_store.rs [hunk 1]
```diff
@@ -5769,6 +5734,71 @@ mod tests {
...
+            assert_eq!(id, expected_id);
+            assert_eq!(record.as_ref(), expected.as_ref());
+        }
+
+        // Re-storing the same ids upserts instead of conflicting.
+        let updated: Vec<(u32, Bytes)> = (1..=4u32)
+            .map(|id| (id, Bytes::from(vec![0xAA; 16])))
+            .collect();
+        store
+            .store_prekeys_batch(&updated, true)
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply storages/sqlite-storage/src/sqlite_store.rs:5786]: Fixed: the test now reads the persisted `uploaded` flag back via `read_query` — `false` after the initial store, `true` after the upsert.
